### PR TITLE
Added: Candidate assessment 

### PR DESCRIPTION
**Pull request by anuragP04**

### CODE REVIEW

File: app/components/assessment/CandidateInstruction.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            )
+          })}
+        </div>
+        {/* Instruction */}
+        <div className="flex flex-col gap-4">
+          <h1 className="mt-8 text-2xl font-semibold leading-8 text-gray-800">
+            {candidateExam.instructions}
+          </h1>
+          <ul className="list-disc">
+            <li>You cannot traverse the sections.</li>
```
Comment (richa-copods) [inline app/components/assessment/CandidateInstruction.tsx:69]: Please create constants for these two list items' text

File: app/components/assessment/CandidateRegister.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const [isChecked, setIsChecked] = useState(false)
+
+  const canSubmitBtnBeEnabled = () => {
+    return firstName.length > 0 && lastName.length > 0 && isChecked
+  }
+
+  const checkBoxProps = {
+    name: 'candidateAgreement',
+    isChecked: false,
+    handleChange: function (event: any) {
```
Comment (richa-copods) [inline app/components/assessment/CandidateRegister.tsx:21]: Please give proper type

File: app/components/assessment/CandidateRegister.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  const inputFieldsProps = [
+    {
+      label: 'First Name',
+      placeholder: 'Enter first name',
+      type: 'text',
+      name: 'firstName',
+      required: true,
+      value: firstName,
+      errorId: 'name-error',
+      onChange: function (event: any) {
```
Comment (richa-copods) [inline app/components/assessment/CandidateRegister.tsx:35]: Please give proper type

File: app/components/assessment/CandidateRegister.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    },
+    {
+      label: 'Last Name',
+      placeholder: 'Enter last name',
+      type: 'text',
+      name: 'lastName',
+      required: true,
+      value: lastName,
+      errorId: 'name-error',
+      onChange: function (event: any) {
```
Comment (richa-copods) [inline app/components/assessment/CandidateRegister.tsx:47]: Please give proper type

File: app/components/assessment/CandidateRegister.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        setLastName(event?.target.value)
+      },
+    },
+  ]
+
+  return (
+    <div className="z-10 flex	min-h-[480px] w-full max-w-[554px] flex-col items-center justify-center rounded-2xl bg-white px-24 drop-shadow-xl">
```
Comment (richa-copods) [inline app/components/assessment/CandidateRegister.tsx:54]: Please remove extra space

File: app/components/assessment/EndAssessment.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+function EndAssesment() {
```
Comment (richa-copods) [inline app/components/assessment/EndAssessment.tsx:1]: Please use arrow function

File: app/components/assessment/EndAssessment.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  return (
+    <div className="flex flex h-screen items-center justify-center">
+      Your assesment is ended
```
Comment (richa-copods) [inline app/components/assessment/EndAssessment.tsx:4]: Please create constant for this

File: app/components/assessment/Question.tsx [hunk 8]
```diff
@@ -0,0 +1,189 @@
+import { useLoaderData } from '@remix-run/react'
+import { useState } from 'react'
+import {
+  candidateExam,
+  commonConstants,
+  QuestionTypes,
+} from '~/constants/common.constants'
+
+function Question() {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:9]: Please use arrow function

File: app/components/assessment/Question.tsx [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+  const { question, section, lastSection } = useLoaderData()
+
+  const questionType = question.question.questionType.value
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:12]: Please use optional chaining

File: app/components/assessment/Question.tsx [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+  const [userAnswer, setUserAnswer] = useState(
+    questionType === QuestionTypes.singleChoice
+      ? question.selectedOptions[0]?.id
+      : questionType == 'TEXT'
+      ? question?.answers
+      : {}
+  )
+
+  if (questionType === QuestionTypes.multipleChoice) {
+    // for (let option of question.question.options) {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:23]: Please remove this if not needed, else add logic

File: app/components/assessment/Question.tsx [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+    //     } else {
+    //       option.rightAnswer = false
+    //     }
+    //   })
+    //   // option.rightAnswer =
+    //   //   question.selectedOptions.indexOf(option.id) == -1 ? false : true
+    // }
+  }
+
+  const onChangeHandle = (event: any, index?: number) => {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:37]: Please give proper type

File: app/components/assessment/Question.tsx [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+    if (questionType == QuestionTypes.singleChoice) {
+      setUserAnswer(event.id)
+    }
+    if (questionType == QuestionTypes.multipleChoice) {
+      // question.options.forEach((opt: any) => {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:42]: Please remove this else add logic

File: app/components/assessment/Question.tsx [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+      // let selectedOption: Array<string> = []
+      // for (let opt of questionData.question.options) {
+      //   if (opt.id == option) {
+      //     opt.rightAnswer = opt.rightAnswer == 'true' ? 'false' : 'true'
+      //     (opt.rightAnswer)
+      //   }
+      // }
+      // setUserAnswer(selectedOption)
+    }
+    if (questionType == QuestionTypes.text) {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:56]: Please use strict check `===`

File: app/components/assessment/Question.tsx [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+function Question() {
+  const { question, section, lastSection } = useLoaderData()
+
+  const questionType = question.question.questionType.value
+
+  const [userAnswer, setUserAnswer] = useState(
+    questionType === QuestionTypes.singleChoice
+      ? question.selectedOptions[0]?.id
+      : questionType == 'TEXT'
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:17]: Please use strict check `===`, also use enum here `QuestionTypes.text`

File: app/components/assessment/Question.tsx [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+    //       option.rightAnswer = false
+    //     }
+    //   })
+    //   // option.rightAnswer =
+    //   //   question.selectedOptions.indexOf(option.id) == -1 ? false : true
+    // }
+  }
+
+  const onChangeHandle = (event: any, index?: number) => {
+    if (questionType == QuestionTypes.singleChoice) {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:38]: Please use strict check

File: app/components/assessment/Question.tsx [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+      setUserAnswer(event.id)
+    }
+    if (questionType == QuestionTypes.multipleChoice) {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:41]: Please use strict check

File: app/components/assessment/Question.tsx [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+      // for (let opt of questionData.question.options) {
+      //   if (opt.id == option) {
+      //     opt.rightAnswer = opt.rightAnswer == 'true' ? 'false' : 'true'
+      //     (opt.rightAnswer)
+      //   }
+      // }
+      // setUserAnswer(selectedOption)
+    }
+    if (questionType == QuestionTypes.text) {
+      setUserAnswer((oldval: Array<string>) => {
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:57]: Please use camelCase for naming parameters

File: app/components/assessment/Question.tsx [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+    }
+  }
+
+  return (
+    <form method="post" className="flex h-full flex-col gap-9">
+      <div className="flex max-h-full flex-1 gap-9 overflow-auto">
+        <div className="flex h-full w-1/2 flex-col gap-3">
+          <div className="flex h-10 items-center justify-between">
+            <div className="flex gap-5 text-lg font-semibold">
+              <span>Question</span>{' '}
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:70]: Please add space properly

File: app/components/assessment/Question.tsx [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+              <span>
+                {question?.order}/{section?.totalQuestions}
+              </span>
+            </div>
+          </div>
+          <div className="shadow-base h-full flex-1 overflow-auto rounded-lg border border-gray-200 bg-white p-4">
+            <div
+              dangerouslySetInnerHTML={{
+                __html: question.question.question,
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:79]: Please use optional chaining if needed, also make sure question is sanitised

File: app/components/assessment/Question.tsx [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+              }}
+            />
+          </div>
+        </div>
+        <div className="flex h-full w-1/2 flex-col gap-3">
+          <div className="flex h-10 items-center justify-between">
+            <div className="flex gap-5 text-lg font-semibold">Options</div>
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:86]: Please create constant for `Options`

File: app/components/assessment/Question.tsx [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+          <div className="shadow-base flex h-full flex-1 flex-col gap-6 overflow-auto rounded-lg border border-gray-200 bg-white p-4">
+            {question?.question?.options?.map(
+              (option: {
+                id: string
+                option: string
+                rightAnswer: boolean
+              }) => {
+                return (
+                  <div key={option.id} className="flex gap-4">
+                    {questionType == 'SINGLE_CHOICE' ? (
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:97]: Please use enum `QuestionTypes.singleChoice` here, also strict checking `===`

File: app/components/assessment/Question.tsx [hunk 22]
```diff
@@ -0,0 +1,189 @@
...
+                      <input
+                        type="radio"
+                        name="option"
+                        value={option.id}
+                        checked={option.id == userAnswer}
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:102]: Please use strict check

File: app/components/assessment/Question.tsx [hunk 23]
```diff
@@ -0,0 +1,189 @@
...
+            )}
+            {question?.question?.correctAnswer?.map(
+              (answer: { id: string }, index: number) => {
+                return (
+                  <div key={answer.id}>
+                    <textarea
+                      name="answer"
+                      id=""
+                      value={userAnswer[index]}
+                      rows={4}
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:130]: Are you sure the number of rows will always be 4?

File: app/components/assessment/Question.tsx [hunk 24]
```diff
@@ -0,0 +1,189 @@
...
+                    <div dangerouslySetInnerHTML={{ __html: option.option }} />
+                  </div>
+                )
+              }
+            )}
+            {question?.question?.correctAnswer?.map(
+              (answer: { id: string }, index: number) => {
+                return (
+                  <div key={answer.id}>
+                    <textarea
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:126]: This tag can be self-closing

File: app/components/assessment/Question.tsx [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+            >
+              {commonConstants.nextButton}
+            </button>
+          ) : lastSection ? (
+            <button
+              className="h-11 w-40 rounded-md border border-primary bg-primary text-base font-medium text-gray-50 shadow-sm"
+              name="endExam"
+              value={section.order}
+              type="submit"
+              disabled={question.order != section.totalQuestions}
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:168]: Please use strict check `!==`

File: app/components/assessment/Question.tsx [hunk 26]
```diff
@@ -0,0 +1,189 @@
...
+            >
+              {candidateExam.endTest}
+            </button>
+          ) : (
+            <button
+              className="h-11 w-40 rounded-md border border-primary bg-primary text-base font-medium text-gray-50 shadow-sm"
+              name="nextSection"
+              value={section.order}
+              type="submit"
+              disabled={question.order != section.totalQuestions}
```
Comment (richa-copods) [inline app/components/assessment/Question.tsx:178]: Please use strict check !==

File: app/components/assessment/SectionDetails.tsx [hunk 27]
```diff
@@ -0,0 +1,47 @@
+import { useLoaderData, useSubmit } from '@remix-run/react'
+import { candidateExam } from '~/constants/common.constants'
+
+function SectionDetails() {
```
Comment (richa-copods) [inline app/components/assessment/SectionDetails.tsx:4]: Please use arrow function

File: app/components/assessment/SectionDetails.tsx [hunk 28]
```diff
@@ -0,0 +1,47 @@
...
+          <div className="flex-1">{section.totalQuestions}</div>
+        </div>
+      </div>
+      <br />
+
+      <button
+        className="rounded-full bg-primary py-2 px-5 text-white"
+        onClick={() => startSection()}
+      >
+        Start Section
```
Comment (richa-copods) [inline app/components/assessment/SectionDetails.tsx:41]: Please create constant for this

File: app/components/assessment/SectionQuestionPage.tsx [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+import { useEffect, useState } from 'react'
+import { candidateExam } from '~/constants/common.constants'
+import { getTimeLeftInSeconds } from '~/utils/assessment.utils'
+
+export default function SectionQuestionPage({
+  section,
+  params,
+  candidateTest,
+}: {
+  section: any
```
Comment (richa-copods) [inline app/components/assessment/SectionQuestionPage.tsx:11]: Please give proper type, same for next two lines

File: app/components/assessment/SectionQuestionPage.tsx [hunk 30]
```diff
@@ -0,0 +1,73 @@
+import { Outlet, useSubmit } from '@remix-run/react'
+import { useEffect, useState } from 'react'
+import { candidateExam } from '~/constants/common.constants'
+import { getTimeLeftInSeconds } from '~/utils/assessment.utils'
+
+export default function SectionQuestionPage({
```
Comment (richa-copods) [inline app/components/assessment/SectionQuestionPage.tsx:6]: Please use arrow function

File: app/components/assessment/SectionQuestionPage.tsx [hunk 31]
```diff
@@ -0,0 +1,73 @@
...
+  section,
+  params,
+  candidateTest,
+}: {
+  section: any
+  params: any
+  candidateTest: any
+}) {
+  const [time, setTimer] = useState(-1)
+  let candidateSection: any = {}
```
Comment (richa-copods) [inline app/components/assessment/SectionQuestionPage.tsx:16]: Please give proper type

File: app/components/assessment/SectionQuestionPage.tsx [hunk 32]
```diff
@@ -0,0 +1,73 @@
...
+  for (let sec of candidateTest?.sections) {
+    if (section?.section?.id == sec?.section?.id) {
+      candidateSection = sec
+      break
+    }
+  }
+  const submit = useSubmit()
+  let timer: any = null
```
Comment (richa-copods) [inline app/components/assessment/SectionQuestionPage.tsx:24]: Please give proper type, avoid using `any`

File: app/components/assessment/SectionQuestionPage.tsx [hunk 33]
```diff
@@ -0,0 +1,73 @@
...
+    let seconds = time - min * 60
+    return `${min > 9 ? '' : '0'}${min}:${seconds > 9 ? '' : '0'}${seconds} `
+  }
+  return (
+    <div className="flex h-full flex-col gap-9">
+      <div className="flex flex-row items-center justify-between">
+        <h1 className="text-2xl font-semibold text-gray-800">
+          {section.section.name}
+        </h1>
+        {/* {params?.questionId && ( */}
```
Comment (richa-copods) [inline app/components/assessment/SectionQuestionPage.tsx:57]: Please remove this

File: app/components/assessment/SectionQuestionPage.tsx [hunk 34]
```diff
@@ -0,0 +1,73 @@
...
+        <div className="flex items-center gap-4 text-lg font-medium text-gray-800">
+          <span className="text-sm font-medium text-gray-500">
+            {candidateExam.timeRemaining}
+          </span>{' '}
+          <span className="flex w-36 justify-center rounded-lg bg-blue-100 px-4 py-1 text-lg font-bold text-gray-800">
+            {getFormattedTime(time)} mins
+          </span>
+        </div>
+        {/* )} */}
```
Comment (richa-copods) [inline app/components/assessment/SectionQuestionPage.tsx:66]: Please remove this

File: app/components/form/Button.tsx [hunk 35]
```diff
@@ -3,7 +3,8 @@ import type { ButtonProps } from '~/interface/Interface'
 export default function Button({ buttonText, ...props }: ButtonProps) {
```
Comment (richa-copods) [inline app/components/form/Button.tsx:3]: Please use arrow function

File: app/components/layouts/CandidateLayout.tsx [hunk 36]
```diff
@@ -0,0 +1,22 @@
+import CandidateSideNav from '../side-nav/CandidateSideNav'
+
+export default function CandidateLayout({
+  children,
+  candidate,
+  candidateTest,
+}: {
+  children: JSX.Element
+  candidate: any
```
Comment (richa-copods) [inline app/components/layouts/CandidateLayout.tsx:9]: Please give proper type, same for next line

File: app/components/side-nav/CandidateSideNav.tsx [hunk 37]
```diff
@@ -0,0 +1,160 @@
+// import logo from '~/../public/assets/logo.svg'
+import { Icon } from '@iconify/react'
+import { useLoaderData, useLocation } from '@remix-run/react'
+import candidateLogo from '~/../public/assets/candidateLogo.svg'
+import Divider from '../divider'
+
+function CandidateSideNav({ candidate, candidateTest }: any) {
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:7]: Please use arrow function and give proper type to props

File: app/components/side-nav/CandidateSideNav.tsx [hunk 38]
```diff
@@ -0,0 +1,160 @@
...
+        </div>
+      </div>
+      <Divider height="2px" />
+      <div className="flex flex-col gap-1 p-5">
+        <div className="text-base font-semibold text-gray-900">
+          {candidateTest?.test?.name}
+        </div>
+        <div className="flex items-center gap-1 text-sm font-medium text-gray-600">
+          <Icon icon={'ic:outline-access-time'} className="text-base" />
+          <span>Time Limit:</span>
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:36]: Please create constant for this

File: app/components/side-nav/CandidateSideNav.tsx [hunk 39]
```diff
@@ -0,0 +1,160 @@
...
+        </div>
+      </div>
+      <Divider height="1px" />
+
+      {/* Candidate Sidenav */}
+      <div className="flex flex-1 flex-col gap-6 py-6">
+        {/* sidenav menu group */}
+        <div className="flex flex-col gap-2.5">
+          <div className="px-5 text-sm font-semibold text-gray-900">
+            Assessment Details
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:47]: Please create constant for this

File: app/components/side-nav/CandidateSideNav.tsx [hunk 40]
```diff
@@ -0,0 +1,160 @@
...
+          </div>
+          <div
+            className={`flex h-11 items-center text-sm  ${
+              location.pathname.includes('/instructions')
+                ? 'border border-0 border-l-4 border-primary bg-blue-50 px-4 font-semibold text-primary'
+                : 'px-5 font-medium text-gray-800 '
+            }`}
+          >
+            Info
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:56]: Please create constant for this

File: app/components/side-nav/CandidateSideNav.tsx [hunk 41]
```diff
@@ -0,0 +1,160 @@
...
+          </div>
+        </div>
+
+        <div className="flex flex-col gap-2.5">
+          <div className="px-5 text-sm font-semibold text-gray-900">
+            Assessment Tests
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:62]: Please create constant for this

File: app/components/side-nav/CandidateSideNav.tsx [hunk 42]
```diff
@@ -0,0 +1,160 @@
...
+          </div>
+          <div>
+            {candidateTest?.sections.map((section: any) => {
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:65]: Please give proper type

File: app/components/side-nav/CandidateSideNav.tsx [hunk 43]
```diff
@@ -0,0 +1,160 @@
...
+              return (
+                <div key={section.id}>
+                  <div
+                    className={`flex h-11 items-center justify-between gap-2 text-sm  ${
+                      routeData?.currentSectionInTest?.section.id ==
+                        section.section.id && !routeData.params?.questionId
+                        ? 'border border-0 border-l-4 border-primary bg-blue-50 px-4 font-semibold text-primary'
+                        : 'px-5 font-medium text-gray-800 '
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:73]: Please remove extra space

File: app/components/side-nav/CandidateSideNav.tsx [hunk 44]
```diff
@@ -0,0 +1,160 @@
...
+                    <span>{section?.section?.name}</span>
+                    {!section?.startedAt &&
+                      routeData?.currentSectionInTest?.section.id !=
+                        section.section.id && (
+                        <Icon
+                          icon={'bytesize:lock'}
+                          className="text-lg text-gray-800"
+                        />
+                      )}
+                    {/* {<Icon
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:85]: Please remove this if not needed anymore

File: app/components/side-nav/CandidateSideNav.tsx [hunk 45]
```diff
@@ -0,0 +1,160 @@
...
+                      <Icon
+                        icon={'teenyicons:tick-circle-outline'}
+                        className="text-lg text-green-800"
+                      />
+                    )}
+                  </div>
+                  {routeData?.currentSectionInTest?.section.id ==
+                    section.section.id &&
+                    routeData.params?.questionId &&
+                    section?.questions.map((question: any) => {
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:99]: Please give proper type

File: app/components/side-nav/CandidateSideNav.tsx [hunk 46]
```diff
@@ -0,0 +1,160 @@
...
+                      return (
+                        <div
+                          key={question.id}
+                          className={`flex h-11 items-center justify-between gap-2 text-sm ${
+                            routeData.params?.questionId == question?.id
+                              ? 'border border-0 border-l-4 border-primary bg-blue-50 pl-7 pr-8 font-semibold text-primary'
+                              : 'px-8 font-medium text-gray-800 '
+                          }`}
+                        >
+                          <span>Question {question?.order}</span>
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:109]: Please use constant for `Question`

File: app/components/side-nav/CandidateSideNav.tsx [hunk 47]
```diff
@@ -0,0 +1,160 @@
...
+                          {question.status == 'SKIPPED' && (
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:110]: Please create enum for question's status

File: app/components/side-nav/CandidateSideNav.tsx [hunk 48]
```diff
@@ -0,0 +1,160 @@
...
+                </div>
+              )
+            })}
+          </div>
+        </div>
+      </div>
+
+      <div className="justify-end p-5">
+        <div>
+          <hr className="mb-3 mt-3 border border-solid border-gray-300"></hr>
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:134]: This tag is self-closing

File: app/components/side-nav/CandidateSideNav.tsx [hunk 49]
```diff
@@ -0,0 +1,160 @@
...
+          <div className="flex items-center justify-between gap-1">
+            <div className="flex items-center gap-1">
+              <div className="flex h-10 w-10 items-center justify-center rounded-full bg-primary">
+                <span className="text-lg font-medium leading-7 text-white">
+                  {candidate.firstName.slice(0, 1)}
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:139]: Please use optional chaining if needed, same for next line

File: app/components/side-nav/CandidateSideNavItems.tsx [hunk 50]
```diff
@@ -0,0 +1,26 @@
+export interface tabProps {
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNavItems.tsx:1]: Please move it to interfaces file

File: app/components/side-nav/CandidateSideNavItems.tsx [hunk 51]
```diff
@@ -0,0 +1,26 @@
...
+  itemName: string
+}
+
+const CandidateSideNavItems = ({ itemName }: tabProps) => {
+  return (
+    <div>
+      <div id="menuItem">
+        <div className="flex flex-row items-start gap-2 rounded-lg bg-blue-50 p-3.5 ">
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNavItems.tsx:9]: Please remove extra space

File: app/components/side-nav/CandidateSideNavItems.tsx [hunk 52]
```diff
@@ -0,0 +1,26 @@
...
+          <div className="flex flex-row items-center gap-2">
+            <span>
+              <p
+                id={itemName}
+                className="non-italic text-sm font-bold	 leading-6
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNavItems.tsx:14]: Please remove extra space..... better remove tailwind from prettierignore file

File: app/components/side-nav/SideNav.tsx [hunk 53]
```diff
@@ -49,13 +49,14 @@ let sideNavGuide = [
     ],
   },
 ]
+const title = 'Quiz'
 
 function SideNav() {
```
Comment (richa-copods) [inline app/components/side-nav/SideNav.tsx:54]: Please use arrow function

File: app/components/tests/SelectSectionCard.tsx [hunk 54]
```diff
@@ -19,7 +19,7 @@ const SelectSectionCard = ({
       toast.error('Cannot add section with 0 questions')
       return
     }
-    var temp = {
+    var tempSection = {
```
Comment (richa-copods) [inline app/components/tests/SelectSectionCard.tsx:22]: Please use `let` instead of `var`

File: app/components/side-nav/SideNav.tsx [hunk 55]
```diff
@@ -49,13 +49,14 @@ let sideNavGuide = [
     ],
   },
 ]
+const title = 'Quiz'
```
Comment (richa-copods) [inline app/components/side-nav/SideNav.tsx:52]: We already have a constant for this in constants file, `export const sideNav = {
  sideNavHeading: 'Quiz',
}` please use this if possible

File: app/constants/common.constants.ts [hunk 56]
```diff
@@ -67,7 +69,23 @@ export const routeFiles = {
...
-  text= "TEXT"
+  multipleChoice = "MULTIPLE_CHOICE",
+  singleChoice = "SINGLE_CHOICE",
+  text = "TEXT"
 }
+
+
+export const candidateExam = {
+  time: "Time",
+  timeRemaining: "Time Remaning",
```
Comment (richa-copods) [inline app/constants/common.constants.ts:80]: Please correct the spelling of `Remaining`

File: app/constants/common.constants.ts [hunk 57]
```diff
@@ -67,7 +69,23 @@ export const routeFiles = {
...
+  minuets: "Mins",
```
Comment (richa-copods) [inline app/constants/common.constants.ts:81]: Please correct the spelling of `minutes`

File: app/models/candidate.server.ts [hunk 58]
```diff
@@ -55,12 +58,13 @@ export async function updateTestLink({ id, link }: { id: string, link: string })
   }
 }
 
-export async function createSectionInTest({ sectionId, candidateTestId, totalQuestions }: { sectionId: string, candidateTestId: string, totalQuestions: number }) {
+export async function createSectionInTest({ sectionId, candidateTestId, order, totalQuestions }: { sectionId: string, candidateTestId: string, order: number, totalQuestions: number }) {
```
Comment (richa-copods) [inline app/models/candidate.server.ts:61]: Please use arrow functions....for entire file

File: app/routes/assessment.tsx [hunk 59]
```diff
@@ -0,0 +1,5 @@
+import { Outlet } from '@remix-run/react'
+
+export default function CandidateAssessment() {
```
Comment (richa-copods) [inline app/routes/assessment.tsx:3]: Please use arrow function

File: app/routes/assessment/$assessmentId/$sectionId.tsx [hunk 60]
```diff
@@ -0,0 +1,98 @@
...
+  return {
+    section,
+    candidateTests,
+    candidate,
+    params,
+    currentSectionInTest,
+  }
+}
+
+// export const action: ActionFunction = async ({ params, request }) => {
```
Comment (richa-copods) [inline app/routes/assessment/$assessmentId/$sectionId.tsx:61]: Please remove if not needed

File: app/routes/assessment/$assessmentId/$sectionId.tsx [hunk 61]
```diff
@@ -0,0 +1,98 @@
...
+//     nextRoute: 'section',
+//     isSection: true,
+//     currentSectionId: nextSectionObject?.id,
+//   })
+
+//   // return redirect(`/assessment/cl70h88f265421bk272f15zgj`)
+//   return redirect(`/assessment/${params.assessmentId}/${nextSectionObject?.id}`)
+// }
+
+export default function AssessmentSection() {
```
Comment (richa-copods) [inline app/routes/assessment/$assessmentId/$sectionId.tsx:86]: Please use arrow function

File: app/routes/assessment/$assessmentId/$sectionId/$questionId.tsx [hunk 62]
```diff
@@ -0,0 +1,77 @@
...
+      params.sectionId as string
+    )
+  }
+
+  return redirect(
+    `/assessment/${params.assessmentId}/${params.sectionId}/${nextQuestionId}`
+  )
+}
+
+export default function AssessmentQuestionForSection() {
```
Comment (richa-copods) [inline app/routes/assessment/$assessmentId/$sectionId/$questionId.tsx:75]: Please use arrow function

File: app/routes/assessment/$assessmentId/instructions.tsx [hunk 63]
```diff
@@ -0,0 +1,71 @@
...
+      currentSectionId: firstSectionId,
+    })
+
+    await startTest(params.assessmentId as string)
+
+    return redirect(`/assessment/${params.assessmentId}/${firstSectionId}`)
+  }
+}
+
+export default function TestInstructions() {
```
Comment (richa-copods) [inline app/routes/assessment/$assessmentId/instructions.tsx:63]: Please use arrow function

File: app/routes/assessment/$assessmentId/register.tsx [hunk 64]
```diff
@@ -0,0 +1,57 @@
...
+        isSection: false,
+        currentSectionId: null,
+      })
+      return redirect(`/assessment/${params.assessmentId}/instructions`)
+    }
+  }
+  return null
+}
+
+export default function CandidateOnboard() {
```
Comment (richa-copods) [inline app/routes/assessment/$assessmentId/register.tsx:51]: Please use arrow function

File: app/routes/assessment/$candidateTestId.tsx [hunk 65]
```diff
@@ -0,0 +1,20 @@
...
+import { getCandidate } from '~/models/assessment.server'
+import { candidateTest } from '~/utils/assessment.utils'
+
+export const loader: LoaderFunction = async ({ params, request }) => {
+  const candidateTests = await candidateTest(params.assessmentId as string)
+  const candidate = await getCandidate(params.assessmentId as string)
+  return { candidateTests, candidate }
+}
+
+export default function CandidateAssessment() {
```
Comment (richa-copods) [inline app/routes/assessment/$candidateTestId.tsx:13]: Please use arrow function

File: app/routes/assessment/$candidateTestId.tsx [hunk 66]
```diff
@@ -0,0 +1,20 @@
...
+  const { candidate, candidateTests } = useLoaderData()
+  return (
+    <CandidateLayout candidate={candidate} candidateTest={candidateTests}>
+      <div>Assessment Candidate Works.....!</div>
```
Comment (richa-copods) [inline app/routes/assessment/$candidateTestId.tsx:17]: Please create a constant for this

File: app/routes/assessment/index.tsx [hunk 67]
```diff
@@ -0,0 +1,5 @@
+export default function CandidateAssessment() {
+  return (
+    <div>Welcome...! You are on wrong page, please visit correct link.</div>
```
Comment (richa-copods) [inline app/routes/assessment/index.tsx:3]: Please create a constant for this

File: app/utils/assessment.utils.ts [hunk 68]
```diff
@@ -0,0 +1,293 @@
+import { redirect } from "@remix-run/server-runtime"
+import { candidateSectionStart, candidateTestStart, checkIfTestLinkIsValid, endAssessment, endCurrentSection, getCandidate, getCandidateIDFromAssessmentID, getCandidateSectionDetails, getCandidateTest, getOrderedSection, getTestInstructionForCandidate, getTestSectionDetails, skipAnswerAndNextQuestion, startAndGetQuestion, updateCandidateFirstLastName, updateNextCandidateStep } from "~/models/assessment.server"
+
+
+// Type of candidateStep field in CandidateTest schema
+export type CandidateStep = {
+  nextRoute: string
+  isSection: boolean
+  currentSectionId: string | null
```
Comment (richa-copods) [inline app/utils/assessment.utils.ts:9]: Please mark it optional, if this can be null

File: app/utils/assessment.utils.ts [hunk 69]
```diff
@@ -0,0 +1,293 @@
...
+  const nextSectionObject = await getSectionByOrder(
+    candidateTest?.test.id as string,
+    order + 1
+  )
+
+  await updateNextStep({
+    assessmentId: assessmentId as string,
+    nextRoute: 'section',
+    isSection: true,
+    currentSectionId: nextSectionObject?.id || null,
```
Comment (richa-copods) [inline app/utils/assessment.utils.ts:235]: Mark it optional if this could be optional

File: prisma/migrations/20220726072920_/migration.sql [hunk 70]
```diff
@@ -0,0 +1,2 @@
+-- AlterTable
+ALTER TABLE "CandidateTest" ADD COLUMN     "candidateStep" JSONB NOT NULL DEFAULT '{ "nextRoute": "register", "isSection": false, "currentSectionID": null }';
```
Comment (richa-copods) [inline prisma/migrations/20220726072920_/migration.sql:2]: Please remove extra space

File: prisma/migrations/20220819051912_add_order_to_question/migration.sql [hunk 71]
```diff
@@ -0,0 +1,33 @@
...
+
+*/
+-- DropForeignKey
+ALTER TABLE "Option" DROP CONSTRAINT "Option_candidateQuestionId_fkey";
+
+-- AlterTable
+ALTER TABLE "Option" DROP COLUMN "candidateQuestionId";
+
+-- AlterTable
+ALTER TABLE "SectionInCandidateTest" ADD COLUMN     "order" INTEGER NOT NULL;
```
Comment (richa-copods) [inline prisma/migrations/20220819051912_add_order_to_question/migration.sql:15]: Please remove extra space

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/assessment/CandidateInstruction.tsx (hunk 1 above)
Comment (anuragP04) [reply app/components/assessment/CandidateInstruction.tsx:69]: This text will be change

File: app/components/side-nav/CandidateSideNav.tsx [hunk 72]
```diff
@@ -0,0 +1,171 @@
...
+                          className={`flex h-11 items-center justify-between gap-2 text-sm ${
+                            routeData.params?.questionId == question?.id
+                              ? 'border border-0 border-l-4 border-primary bg-blue-50 pl-7 pr-8 font-semibold text-primary'
+                              : 'px-8 font-medium text-gray-800 '
+                          }`}
+                        >
+                          <span>
+                            {candidateExam.question} {question?.order}
+                          </span>
+                          {question.status == QuestionStatus.skipped && (
```
Comment (richa-copods) [inline app/components/side-nav/CandidateSideNav.tsx:121]: Please use strict check `===`, for entire file

**Review by richa-copods — CHANGES_REQUESTED** [review]

**Review by richa-copods — APPROVED** [review]

**Pull request merged**